From 327b70f32793c34f8b285d52bccda1229aaa2575 Mon Sep 17 00:00:00 2001
From: Rohith Sharma K S <rohithsharmaks@apache.org>
Date: Mon, 13 Jun 2016 11:09:32 +0530
Subject: [PATCH 1808/2748] YARN-4989.
 TestWorkPreservingRMRestart#testCapacitySchedulerRecovery
 fails intermittently. Contributed by Ajith S.

(cherry picked from commit 28b66ae919e348123f4c05a4787c9ec56c087c25)

Change-Id: Iaca1996fe970d16cea675ef292fab1dd591f2713
---
 .../TestWorkPreservingRMRestart.java               |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestWorkPreservingRMRestart.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestWorkPreservingRMRestart.java
index cce232d..578608b 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestWorkPreservingRMRestart.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestWorkPreservingRMRestart.java
@@ -441,7 +441,7 @@ public void testCapacitySchedulerRecovery() throws Exception {
     // Wait for RM to settle down on recovering containers;
     waitForNumContainersToRecover(2, rm2, am1_1.getApplicationAttemptId());
     waitForNumContainersToRecover(2, rm2, am1_2.getApplicationAttemptId());
-    waitForNumContainersToRecover(2, rm2, am1_2.getApplicationAttemptId());
+    waitForNumContainersToRecover(2, rm2, am2.getApplicationAttemptId());
 
     // Calculate each queue's resource usage.
     Resource containerResource = Resource.newInstance(1024, 1);
-- 
1.7.9.5

